msal_extensions/osx.py

```diff
@@ -172,6 +172,8 @@ def get_generic_password(self, service, account_name):
             contents,
             None,
         )
+        if exit_status == KeychainError.ITEM_NOT_FOUND:
+            return None
```

Comment:
> For having it in persistence layer, I did think about that option, however since this is an macOS specific error, I think having it is osx.py makes more sense

The problem of "fixing" it here in osx.py is that, the introduction of a new `None` return value would unnecessarily modify/broaden the meaning of this method. `None` was not an expected outcome of this method, per its original design. And it seems we do not have to change that design here.

Would be great to see an alternative of fixing this in persistence layer and see how it works.